src/history/hs_rec.c

```diff
@@ -1100,21 +1093,20 @@ __hs_delete_record(
         WT_ASSERT(session, tombstone != NULL && __wt_txn_upd_visible_all(session, tombstone));
         ret = 0;
     } else {
-#ifdef HAVE_DIAGNOSTIC
-        __wt_hs_upd_time_window(r->hs_cursor, &hs_tw);
-        WT_ASSERT(session, hs_tw->start_txn == WT_TXN_NONE || hs_tw->start_txn == upd->txnid);
-        WT_ASSERT(session, hs_tw->start_ts == WT_TS_NONE || hs_tw->start_ts == upd->start_ts);
-        WT_ASSERT(session,
-          hs_tw->durable_start_ts == WT_TS_NONE || hs_tw->durable_start_ts == upd->durable_ts);
-        if (tombstone != NULL) {
-            WT_ASSERT(session, hs_tw->stop_txn == tombstone->txnid);
-            WT_ASSERT(session, hs_tw->stop_ts == tombstone->start_ts);
-            WT_ASSERT(session, hs_tw->durable_stop_ts == tombstone->durable_ts);
+        if (DIAGNOSTIC_ASSERTS_ENABLED(session)) {
+            __wt_hs_upd_time_window(r->hs_cursor, &hs_tw);
+            WT_ASSERT(session, hs_tw->start_txn == WT_TXN_NONE || hs_tw->start_txn == upd->txnid);
+            WT_ASSERT(session, hs_tw->start_ts == WT_TS_NONE || hs_tw->start_ts == upd->start_ts);
+            WT_ASSERT(session,
+              hs_tw->durable_start_ts == WT_TS_NONE || hs_tw->durable_start_ts == upd->durable_ts);
+            if (tombstone != NULL) {
+                WT_ASSERT(session, hs_tw->stop_txn == tombstone->txnid);
+                WT_ASSERT(session, hs_tw->stop_ts == tombstone->start_ts);
+                WT_ASSERT(session, hs_tw->durable_stop_ts == tombstone->durable_ts);
+            } else
+                WT_ASSERT(session, !WT_TIME_WINDOW_HAS_STOP(hs_tw));
         } else
```

Comment:
No need for an else here. 
`WT_ERR(r->hs_cursor->remove(r->hs_cursor));` should run regardless of if diagnostic asserts are enabled/disabled.

Reply:
Done in: [ea2caa5](https://github.com/wiredtiger/wiredtiger/pull/8606/commits/ea2caa5383d6da4a6a9580b41f6fe00365e5b088)